modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -307,6 +339,15 @@ public PollItemStatus accept(Consumer<PollItem<T, A>> consumer) {
           status = FILTERED_BY_WATERMARK;
         } else if (currentPollItems < maxItemsPerPoll) {
           currentPollItems++;
+          if (EMIT_NOTIFICATIONS) {
+            callbackContext
+                .addVariable(ACCEPTED_ITEM_INFORMATION,
+                             new ItemInformation(getPollId(), itemId, pollItem.getWatermark(), componentLocation.getLocation()));
+            /*
```

Comment:
remove

Reply:
this is needed to propagate the polling information until the event is created and the notification can be sent. unless we find another mechanism/way, this can't be removed